backend.py

```diff
@@ -1658,8 +1659,11 @@ def writeInventory(installID, controlID, mounts, primary_disk, backup_partnum, s
     inv.write("DOM0_MEM='%d'\n" % host_config['dom0-mem'])
     inv.write("DOM0_VCPUS='%d'\n" % host_config['dom0-vcpus'])
     inv.write("MANAGEMENT_INTERFACE='%s'\n" % admin_bridge)
+    # if no modes are configured then use mgmtAddrType
+    if not (admin_config.mode or admin_config.modev6):
+        inv.write("MANAGEMENT_ADDRESS_TYPE='%s'\n" % mgmtAddrType)
     # Default to IPv4 unless we have only got an IPv6 admin interface
-    if ((not admin_config.mode) and admin_config.modev6):
+    elif (not admin_config.mode) and admin_config.modev6:
```

Comment:
wouldn't it be more readable to have a single `if not admin_config.mode` around all of this?

Reply:
I think that cases are more readable like this but I can modify it no worries.

Reply:
maybe with the updated logic we can still use `elif` to get get things more readable?